test/stability-tests/src/it/java/software/amazon/awssdk/stability/tests/utils/StabilityTestUtils.java

```diff
@@ -0,0 +1,115 @@
+    }
+
+    /**
+     * Process the result. Throws exception if SdkClientExceptions were thrown or 5% requests failed of
+     * SdkServiceException or IOException.
+     *
+     * @param testResult the result to process.
+     */
+    public static void processResult(TestResult testResult) {
+        LOGGER.info(() -> "TestResult: " + testResult);
+
+        int clientExceptionCount = testResult.clientExceptionCount();
+
+        int expectedExceptionCount = testResult.ioExceptionCount() + testResult.serviceExceptionCount();
+
+        double ratio = expectedExceptionCount / (double) testResult.totalRequestCount();
+        if (clientExceptionCount > 0) {
+            throw new RuntimeException(String.format("%s SdkClientExceptions were thrown, failing the tests", clientExceptionCount));
+        }
+
+        if (ratio > ALLOWED_FAILURE_RATIO) {
+            throw new RuntimeException(String.format("More than %s requests failed of SdkServiceException or IOException, failing the tests",
+                                                     ALLOWED_FAILURE_RATIO * 100));
+        }
+    }
+
+    /**
+     * Handle the exceptions of executing the futures.
+     *
+     * @param future the future to be executed
+     * @param exceptionCounter the exception counter
+     * @return the completable future
+     */
+    public static CompletableFuture<?> handleException(CompletableFuture<?> future, ExceptionCounter exceptionCounter) {
+
+        return future.handle((r, t) -> {
+            if (t != null) {
+                if (t instanceof SdkServiceException) {
+                    exceptionCounter.addServiceException();
+                } else if (t instanceof IOException) {
+                    exceptionCounter.addIoException();
+                } else if (t instanceof SdkClientException) {
+                    exceptionCounter.addClientException();
+                } else {
+                    LOGGER.error(() -> "Unknown exception was thrown ", t);
+                    exceptionCounter.addUnknownException();
+                }
+            }
+            return r;
+        });
+    }
+
+
+    /**
+     * Helper method to run the async tests and process the results.
+     *
+     * @param futureFactory the function to provide the CompletableFuture
+     * @param totalRequestNumber the total number of the request
+     * @param testName the name of the tests
+     */
+    public static void runAsyncTests(IntFunction<CompletableFuture<?>> futureFactory, int totalRequestNumber, String testName) {
+        ExceptionCounter exceptionCounter = new ExceptionCounter();
+        CompletableFuture[] completableFutures = new CompletableFuture[totalRequestNumber];
+
+        for (int i = 0; i < totalRequestNumber; i++) {
+            CompletableFuture<?> future = futureFactory.apply(i);
+            completableFutures[i] = handleException(future, exceptionCounter);
+        }
+
+        CompletableFuture.allOf(completableFutures).join();
```

Comment:
Should we add a timeout to catch thinks like futures not being completed?

Reply:
Good call, will add